### PR TITLE
Update URL with #anchor

### LINKED ISSUE #8951: [Docs] Update the URL dynamically according to anchor tag.
### Current State
When you click on a hyperlink that is linked to an anchor, such as a heading, the anchor isn't included in the URL querystring. Consequently, when you attempt to share a specific section of a page by clicking the anchor, then copying the URL from your browser's address bar, the clipboard doesn't capture the specific section location represented by the anchor (#anchor).

### Desired State
The expected outcome for all meshery doc pages is to have URLs dynamically updated when an anchor is clicked and accessible from different locations.

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by goldfishdolphin**
**Notes for Reviewers**

This PR fixes #8951

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 4b63f736236c2e7e3d63b0380c06b78b8c06dfb9 at: https://6515470f9640e97afb908348--meshery-docs-preview.netlify.app

File: docs/_includes/head.html [hunk 1]
```diff
@@ -69,7 +59,12 @@
       };
       anchors.add("h1, h2, h3, h4, h5");
     });
```
Comment (leecalcote) [inline docs/_includes/head.html:61]: Lines 20 through 61 are formatting changes that ideally are excluded. We're in need of enforcement using a common prettier precommit hook, @nebula-aac

File: docs/_includes/head.html [hunk 2]
```diff
@@ -69,7 +59,12 @@
...
+    // Function to handle anchor links
+    document.addEventListener('click', function (event) {
+      if (event.target.tagName === 'A' && event.target.getAttribute('href').startsWith('#')) {
```
Comment (leecalcote) [inline docs/_includes/head.html:64]: @goldfishdolphin is this case insensitive?

Comment (leecalcote) [inline docs/_includes/head.html:64]: @goldfishdolphin is this match of "A" case insensitive? If not, we might need to account for "a", too.

**Review by leecalcote — APPROVED** [review]
The changes in this PR are behaving well in the couple of pages that I tested.

**Comment by l5io** [comment]
🚀 Preview for commit 4b439d7eaa106b52251607e6ab716c1e68475a76 at: https://651663063a3a353749f59b7b--meshery-docs-preview.netlify.app

**Pull request merged**